modules/market/src/main/java/com/opengamma/strata/market/curve/NodalCurve.java

```diff
@@ -80,6 +80,17 @@ public default ParameterMetadata getParameterMetadata(int parameterIndex) {
    */
   public abstract NodalCurve withYValues(DoubleArray values);
 
+  /**
+   * Returns a new curve with the specified x-values and y-values.
+   * <p>
+   * This allows the x values and y-values of the curve to be changed.
+   * 
+   * @param xValues  the new x-values for the curve
+   * @param yValues  the new y-values for the curve
+   * @return the new curve
+   */
+  public abstract NodalCurve withXYValues(DoubleArray xValues, DoubleArray yValues);
```

Comment:
Call this method `withValues()`.
Change `withYValues()` to be a default method that delegates to `withValues(getXValues(), yValues)`;
Both existing ISDA curves should implement the new method.